FigSupplier.java

```diff
@@ -0,0 +1,28 @@
+import java.util.Random;
+
+class FigSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Shape getRandomFigure() {
+        int figureType = random.nextInt(5);

```

Comment:
Consider defining a constant for the number of figure types instead of using the magic number `5`. This will improve code readability and maintainability. For example: `private static final int FIGURE_COUNT = 5;`.